### ISSUE TITLE
The tracked callback should be implemented using after_commit instead of after_create/after_update

### ISSUE THREAD

**Issue by Looooong** [simukappu/activity_notification#99]
Whenever `notify_later` or `send_later` is enabled in the `tracked` option on a notifiable, a job is enqueued to be processed. However, in some instance, the job back-end may process the job before the notifiable is persisted in the database. This results in exception on the job back-end.

I suggest that this code:
https://github.com/simukappu/activity_notification/blob/07f13c1a5acc229471ee650b9437728c818f0dad/lib/activity_notification/roles/acts_as_notifiable.rb#L280-L289

should be replaced with:
```ruby
 def add_tracked_callback(tracked_callbacks, tracked_action, tracked_proc) 
   if tracked_callbacks.include? tracked_action 
     after_commit(tracked_proc, on: tracked_action) 
   end 
 end 
```

**Comment by irondnb** [comment]
Totally agree, as after_commit callback ensures that transaction is finished and object persist in database. Im using override in initializer to avoid this problem.

**Comment by simukappu** [comment]
Can you create a pull request for this?

**Comment by Looooong** [comment]
I opened #100, but couldn't run the test because of dependency issue. Can you check that out?

**Comment by simukappu** [comment]
Thank you for your PR. As you mentioned, we have to pass the tests. It seems to be more than just dependency issue.
https://travis-ci.org/simukappu/activity_notification/builds/504662767

**Comment by simukappu** [comment]
I'm working to fix this dependency issue for testing.

**Comment by simukappu** [comment]
I have fixed this dependency issue for testing in this [commit](https://github.com/simukappu/activity_notification/commit/23b3d50e386b45afc126186150f21f39b9c38a3d).
For now, it is fixed in master and development branches.

@Looooong, can you check this pull request again?

**Comment by simukappu** [comment]
Current test result is here: https://travis-ci.org/simukappu/activity_notification/builds/520095237

**Comment by Looooong** [comment]
@simukappu I have modified the pull requests. And somehow, a few tests only in Rails 5.1 are failed. I wonder if it's Rails specific error?

**Comment by irondnb** [comment]
> @simukappu I have modified the pull requests. And somehow, a few tests only in Rails 5.1 are failed. I wonder if it's Rails specific error?

Actually there was issue with after_commit for 5.1.x. It was fixed on 5.1-stable

https://github.com/rails/rails/issues/30779
https://github.com/rails/rails/pull/32176

**Comment by Looooong** [comment]
@irondnb Thank you for the information. I couldn't get my head around this. Is there anything we can do to solve this problem?

**Comment by simukappu** [comment]
@Looooong I avoided this Rails issue with above commit and will merge your PR. Thank you for your contribution!

**Comment by simukappu** [comment]
Just released as v1.7.1. Thanks!
